### PR TITLE
sql/sqlutil, descs: move `TxnWithExecutor()` and methods to `sql.InternalExecutorFactory`

### PR DESCRIPTION
**Pull request by ZhouXing19**
Having `TxnWithExecutor()` in `descs.CollectionFactory` is unnatural, and will bring dependency loop headaches. This commit is to move the same logic under `sql.InternalExecutorFactory`.

Release note: None

### CODE REVIEW

File: pkg/sql/catalog/descs/txn.go [hunk 1]
```diff
@@ -89,6 +89,66 @@ type TxnWithExecutorFunc = func(
 	ie sqlutil.InternalExecutor,
 ) error
 
+// TxnDB include methods to run transactions with a *Collection.
+type TxnDB interface {
```
Comment (ZhouXing19) [inline pkg/sql/catalog/descs/txn.go:93]: Do we really want this? Wouldn't it be enough to inject `sqlutil.InternalExecutorFactory` and use these methods directly?

**Comment by ZhouXing19** [comment]
RFAL

**Review by ajwerner — COMMENTED** [review]
This seems right. Let's unwind the collection factory methods in the same PR while you're here. 

> Reviewed 3 of 4 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/88875#-:-ND4ZX717R6DlSK-7DPy:b-9ouj0b)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ajwerner and @ZhouXing19)

___
*[`pkg/sql/catalog/descs/txn.go` line 93 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/88875#-ND0Bni78UHBHs3rAGzo-r1-93:-ND4YuJ6CzKbzSDzQHO-:b-m8s3gy) ([raw file](https://github.com/cockroachdb/cockroach/blob/bf0e557d429befe072989acda660a1ded8050e6f/pkg/sql/catalog/descs/txn.go#L93)):*
<details><summary><i>Previously, ZhouXing19 (Jane Xing) wrote…</i></summary><blockquote>

Do we really want this? Wouldn't it be enough to inject `sqlutil.InternalExecutorFactory` and use these methods directly?
</blockquote></details>

Yeah I think you're right that that's better.
___
*[`pkg/sql/catalog/descs/txn.go` line 115 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/88875#-ND4ZJXa2zOHJqeHoOHz:-ND4ZJXa2zOHJqeHoOI-:bcxyt8v) ([raw file](https://github.com/cockroachdb/cockroach/blob/769668ae7d25a9473acc7bdd38e6463f0b6af2a5/pkg/sql/catalog/descs/txn.go#L115)):*
> ```Go
> // WaitForDescriptorsFunc return a function that waits for descriptors that
> // were modified, skipping over ones that had their descriptor wiped.
> func (cf *CollectionFactory) WaitForDescriptorsFunc(
> ```

I don't see why this needs to be here. The `InternalExecutorFactory` implementation has a `leaseMgr`. We're not getting anything else out of this method, are we?



_Code quote:_
```Go
// WaitForDescriptorsFunc return a function that waits for descriptors that
// were modified, skipping over ones that had their descriptor wiped.
func (cf *CollectionFactory) WaitForDescriptorsFunc(
	ctx context.Context,
) WaitForModifiedDescsFunc {
	return func(modifiedDescriptors []lease.IDVersion, deletedDescs catalog.DescriptorIDSet) error {
```

**Review by ZhouXing19 — COMMENTED** [review]
> Let's unwind the collection factory methods in the same PR

Hmm do you mean completely replace `descs.CollectionFactory.TxnWithExecutor()` with `sql.InternalExecutor.TxnWithExecutor()`? I can do that, but want to get your approval first for the new methods we have here, so that we can have a concrete plan to change the existing usages. 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/88875#-:-ND4eGlFBeaR6sTzJ4Il:b6cmbka)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ajwerner)

___
*[`pkg/sql/catalog/descs/txn.go` line 93 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/88875#-ND0Bni78UHBHs3rAGzo-r1-93:-ND4dOaT9IofIfvPx_4y:bavbhh8) ([raw file](https://github.com/cockroachdb/cockroach/blob/bf0e557d429befe072989acda660a1ded8050e6f/pkg/sql/catalog/descs/txn.go#L93)):*
<details><summary><i>Previously, ajwerner wrote…</i></summary><blockquote>

Yeah I think you're right that that's better.
</blockquote></details>

Ah, I gave a second thought and I think you made sense. `DescsTxnWithExecutor()` and `DescsTxn()` are not in `sqlutil`. So if someone wants to use them, they have to convert `sqltuil.InternalExecutor` to `sql.InternalExecutor`, which means they will have to import `sql`. But with these 2 methods declared in an interface in `descs`, they just have to import `descs`, which is, I think, lighter than `sql`. 
So I just moved them to `descs.InternalExecutorFactoryWithTxn` above.
___
*[`pkg/sql/catalog/descs/txn.go` line 115 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/88875#-ND4ZJXa2zOHJqeHoOHz:-ND4e4Jz3IQ7Di9Yv9zr:bd03ml9) ([raw file](https://github.com/cockroachdb/cockroach/blob/769668ae7d25a9473acc7bdd38e6463f0b6af2a5/pkg/sql/catalog/descs/txn.go#L115)):*
<details><summary><i>Previously, ajwerner wrote…</i></summary><blockquote>

I don't see why this needs to be here. The `InternalExecutorFactory` implementation has a `leaseMgr`. We're not getting anything else out of this method, are we?
</blockquote></details>

Right there's no need to separate them into another function. Changed it back.

**Review by ajwerner — COMMENTED** [review]
this is looking right to me 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/88875#-:-ND4is9j-5qySBUe8-Wv:b-1mp4z)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ajwerner and @ZhouXing19)

___
*[`pkg/sql/internal.go` line 1402 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/88875#-ND4ieONA7u_5tkhAe0J:-ND4ieONA7u_5tkhAe0K:b-42gt2r) ([raw file](https://github.com/cockroachdb/cockroach/blob/07cf6a27fde9ccf0a0117a716f74ff1490b9ea2e/pkg/sql/internal.go#L1402)):*
> ```Go
> ) error {
> 	run := ApplyTxnOptions(db, opts...)
> 	cf := ief.server.cfg.CollectionFactory
> ```

I don't think we need to involve the `CollectionFactory` at all here, the lease manager is `ief.server.cfg.LeaseManager`
___
*[`pkg/sql/catalog/descs/txn.go` line 93 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/88875#-ND0Bni78UHBHs3rAGzo-r1-93:-ND4iLlhEgguOsuYan9Z:b4f8jad) ([raw file](https://github.com/cockroachdb/cockroach/blob/bf0e557d429befe072989acda660a1ded8050e6f/pkg/sql/catalog/descs/txn.go#L93)):*
<details><summary><i>Previously, ZhouXing19 (Jane Xing) wrote…</i></summary><blockquote>

Ah, I gave a second thought and I think you made sense. `DescsTxnWithExecutor()` and `DescsTxn()` are not in `sqlutil`. So if someone wants to use them, they have to convert `sqltuil.InternalExecutor` to `sql.InternalExecutor`, which means they will have to import `sql`. But with these 2 methods declared in an interface in `descs`, they just have to import `descs`, which is, I think, lighter than `sql`. 
So I just moved them to `descs.InternalExecutorFactoryWithTxn` above.
</blockquote></details>

right because we don't want `sqlutil` to depend on `descs` :thumbsup:

**Review by ZhouXing19 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/88875)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ajwerner and @ZhouXing19)

___
*[`pkg/sql/internal.go` line 1402 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/88875#-ND4ieONA7u_5tkhAe0J:-ND4kDAlBBYsgSC0L3J1:b-6f23p) ([raw file](https://github.com/cockroachdb/cockroach/blob/07cf6a27fde9ccf0a0117a716f74ff1490b9ea2e/pkg/sql/internal.go#L1402)):*
<details><summary><i>Previously, ajwerner wrote…</i></summary><blockquote>

I don't think we need to involve the `CollectionFactory` at all here, the lease manager is `ief.server.cfg.LeaseManager`
</blockquote></details>

But eventually we'd need `cf.NewCollection()` to create the descriptor collection for the internal executor

**Review by ajwerner — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/88875)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ajwerner and @ZhouXing19)

___
*[`pkg/sql/internal.go` line 1402 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/88875#-ND4ieONA7u_5tkhAe0J:-ND4mHWM4V2jIbyIi4TN:b-hy407t) ([raw file](https://github.com/cockroachdb/cockroach/blob/07cf6a27fde9ccf0a0117a716f74ff1490b9ea2e/pkg/sql/internal.go#L1402)):*
<details><summary><i>Previously, ZhouXing19 (Jane Xing) wrote…</i></summary><blockquote>

But eventually we'd need `cf.NewCollection()` to create the descriptor collection for the internal executor
</blockquote></details>

right right, I guess I just mean we don't need its `GetLeaseManager()` method (which maybe we can now remove?)

**Review by ZhouXing19 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/88875)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ajwerner and @ZhouXing19)

___
*[`pkg/sql/internal.go` line 1402 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/88875#-ND4ieONA7u_5tkhAe0J:-ND4nctW-StMVsLKIX0B:b-348bze) ([raw file](https://github.com/cockroachdb/cockroach/blob/07cf6a27fde9ccf0a0117a716f74ff1490b9ea2e/pkg/sql/internal.go#L1402)):*
<details><summary><i>Previously, ajwerner wrote…</i></summary><blockquote>

right right, I guess I just mean we don't need its `GetLeaseManager()` method (which maybe we can now remove?)
</blockquote></details>

Oh `descs.CollectionFactory.leaseMgr` can't be exported from `descs`, and we're at `sql` here, so i think we still need `GetLeaseManager()`

**Review by ajwerner — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/88875)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ajwerner and @ZhouXing19)

___
*[`pkg/sql/internal.go` line 1402 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/88875#-ND4ieONA7u_5tkhAe0J:-ND4t4a00xDjfJnKDJJW:b-hegxpc) ([raw file](https://github.com/cockroachdb/cockroach/blob/07cf6a27fde9ccf0a0117a716f74ff1490b9ea2e/pkg/sql/internal.go#L1402)):*
<details><summary><i>Previously, ZhouXing19 (Jane Xing) wrote…</i></summary><blockquote>

Oh `descs.CollectionFactory.leaseMgr` can't be exported from `descs`, and we're at `sql` here, so i think we still need `GetLeaseManager()`
</blockquote></details>

`ief.server.cfg.LeaseManager` has a handle to `LeaseManager` already

**Comment by ZhouXing19** [comment]
*[`pkg/sql/internal.go` line 1402 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/88875#-ND4ieONA7u_5tkhAe0J:-ND4tXrK2_ENZsVUgDX9:b-z3bek2) ([raw file](https://github.com/cockroachdb/cockroach/blob/07cf6a27fde9ccf0a0117a716f74ff1490b9ea2e/pkg/sql/internal.go#L1402)):*
<details><summary><i>Previously, ajwerner wrote…</i></summary><blockquote>

`ief.server.cfg.LeaseManager` has a handle to `LeaseManager` already
</blockquote></details>

Ah yeah right 😅 made the change!

**Comment by ZhouXing19** [comment]
Failed tests were flacky, RFAL, thanks!

**Comment by ZhouXing19** [comment]
Friendly re-pinging for a review. Thanks!

**Review by ajwerner — COMMENTED** [review]
In general, can you take a pass at removing the reaching from the server to the collection factory to an internal executor factory and rework those to go from the server right to the InternalExecutorFactory? 

> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/88875#-:-ND9wbUM8RjNVAOWrR7Z:b-fltec0)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ajwerner, @rhu713, and @ZhouXing19)

___
*[`pkg/ccl/backupccl/restore_planning.go` line 709 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/88875#-ND9v4zs5j-Ucxp5nViM:-ND9v4zs5j-Ucxp5nViN:bf1eyog) ([raw file](https://github.com/cockroachdb/cockroach/blob/b46b350b61c8c938820c25ccd62c2fc8ea0694e1/pkg/ccl/backupccl/restore_planning.go#L709)):*
> ```Go
> // restore.
> func dropDefaultUserDBs(ctx context.Context, execCfg *sql.ExecutorConfig) error {
> 	ief := execCfg.InternalExecutorFactory.(*sql.InternalExecutorFactory)
> ```

I don't love this type assertion. Can we define a superset interface for the `sql.ExecutorConfig`. Maybe `descs.InternalExecutorFactory` which embeds `sqlutils.InternalExecutorFactory` and has the added methods?
___
*[`pkg/sql/catalog/descs/factory.go` line 41 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/88875#-ND9w-Jt6z68J-baCE9O:-ND9w-Jt6z68J-baCE9P:b-mqwjh0) ([raw file](https://github.com/cockroachdb/cockroach/blob/b46b350b61c8c938820c25ccd62c2fc8ea0694e1/pkg/sql/catalog/descs/factory.go#L41)):*
> ```Go
> 	spanConfigLimiter  spanconfig.Limiter
> 	defaultMonitor     *mon.BytesMonitor
> 	ieFactoryWithTxn   InternalExecutorFactoryWithTxn
> ```

what is this now used for?
___
*[`pkg/sql/catalog/descs/factory.go` line 52 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/88875#-ND9vfUnAcXXgfK-EZWY:-ND9vfUnAcXXgfK-EZWZ:b9l0a6i) ([raw file](https://github.com/cockroachdb/cockroach/blob/b46b350b61c8c938820c25ccd62c2fc8ea0694e1/pkg/sql/catalog/descs/factory.go#L52)):*
> ```Go
> // collection factory.
> func (cf *CollectionFactory) GetInternalExecutorFactory() InternalExecutorFactoryWithTxn {
> 	return cf.ieFactoryWithTxn
> ```

Why do these exist? The places they seem to be accessed already have the 



_Code quote:_
```Go
// GetClusterSettings returns the cluster setting from the collection factory.
func (cf *CollectionFactory) GetClusterSettings() *cluster.Settings {
	return cf.settings
}

// GetInternalExecutorFactory returns the internal executor factory of the
// collection factory.
func (cf *CollectionFactory) GetInternalExecutorFactory() InternalExecutorFactoryWithTxn {
	return cf.ieFactoryWithTxn
}
```
___
*[`pkg/sql/catalog/descs/factory.go` line 58 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/88875#-ND9w95H5pfDwzU6Ed-0:-ND9w95H5pfDwzU6Ed-1:byw7uao) ([raw file](https://github.com/cockroachdb/cockroach/blob/b46b350b61c8c938820c25ccd62c2fc8ea0694e1/pkg/sql/catalog/descs/factory.go#L58)):*
> ```Go
> // with associated extra txn state information.
> // It should only be used as a field hanging off CollectionFactory.
> type InternalExecutorFactoryWithTxn interface {
> ```

Let's rename this to just `InternalExecutorFactory` and embed `sqlutils.InternalExecutorFactory` in it? Alternatively, what do you think about `TxnManager` as the name?
___
*[`pkg/sql/catalog/descs/factory.go` line 59 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/88875#-ND9vZf50UkroTxdEMDq:-ND9vZf50UkroTxdEMDr:br0ah6n) ([raw file](https://github.com/cockroachdb/cockroach/blob/b46b350b61c8c938820c25ccd62c2fc8ea0694e1/pkg/sql/catalog/descs/factory.go#L59)):*
> ```Go
> // It should only be used as a field hanging off CollectionFactory.
> type InternalExecutorFactoryWithTxn interface {
> 	MemoryMonitor() *mon.BytesMonitor
> ```

Are either of these methods needed anymore?



_Code quote:_
```Go
	MemoryMonitor() *mon.BytesMonitor

	NewInternalExecutorWithTxn(
		sd *sessiondata.SessionData,
		sv *settings.Values,
		txn *kv.Txn,
		descCol *Collection,
	) (sqlutil.InternalExecutor, InternalExecutorCommitTxnFunc)
```
___
*[`pkg/sql/catalog/descs/system_table.go` line 46 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/88875#-ND9vu3GCIEmAMQA6rnl:-ND9vu3GCIEmAMQA6rnm:bgy3il8) ([raw file](https://github.com/cockroachdb/cockroach/blob/b46b350b61c8c938820c25ccd62c2fc8ea0694e1/pkg/sql/catalog/descs/system_table.go#L46)):*
> ```Go
> 
> 	var id descpb.ID
> 	if err := r.collectionFactory.GetInternalExecutorFactory().DescsTxn(ctx, r.db, func(
> ```

let's change the dependency here 
___
*[`pkg/sql/stats/automatic_stats.go` line 354 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/88875#-ND9wL3W9LYQd19yN6U5:-ND9wL3W9LYQd19yN6U6:b-f6575j) ([raw file](https://github.com/cockroachdb/cockroach/blob/b46b350b61c8c938820c25ccd62c2fc8ea0694e1/pkg/sql/stats/automatic_stats.go#L354)):*
> ```Go
> 	ctx context.Context, tableID descpb.ID,
> ) (desc catalog.TableDescriptor) {
> 	if err := r.cache.collectionFactory.GetInternalExecutorFactory().DescsTxn(ctx, r.cache.ClientDB, func(
> ```

let's change the dependency here to `descs.InternalExecutorFactory`

**Review by ZhouXing19 — COMMENTED** [review]
> In general, can you take a pass at removing the reaching from the server to the collection factory to an internal executor factory and rework those to go from the server right to the InternalExecutorFactory?

Made the changes! 
This should be ready for another look. Thank you!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/88875#-:-NDC4c4c02I7uUM86_CT:bgkw4m0)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ajwerner, @herkolategan, @rhu713, and @ZhouXing19)

___
*[`pkg/ccl/backupccl/restore_planning.go` line 709 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/88875#-ND9v4zs5j-Ucxp5nViM:-NDC4-CJ4O4xFhsjZc76:b-7ma734) ([raw file](https://github.com/cockroachdb/cockroach/blob/b46b350b61c8c938820c25ccd62c2fc8ea0694e1/pkg/ccl/backupccl/restore_planning.go#L709)):*
<details><summary><i>Previously, ajwerner wrote…</i></summary><blockquote>

I don't love this type assertion. Can we define a superset interface for the `sql.ExecutorConfig`. Maybe `descs.InternalExecutorFactory` which embeds `sqlutils.InternalExecutorFactory` and has the added methods?
</blockquote></details>

Thanks for the good idea! I made `descs.TxnManager` a superset interface, and changed `execCfg.InternalExecutorFactory` to a `descs.TxnManager` as well.
___
*[`pkg/sql/catalog/descs/factory.go` line 41 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/88875#-ND9w-Jt6z68J-baCE9O:-NDC4Lnw5w99mUnh0Jyh:bss4a2b) ([raw file](https://github.com/cockroachdb/cockroach/blob/b46b350b61c8c938820c25ccd62c2fc8ea0694e1/pkg/sql/catalog/descs/factory.go#L41)):*
<details><summary><i>Previously, ajwerner wrote…</i></summary><blockquote>

what is this now used for?
</blockquote></details>

Removed.
___
*[`pkg/sql/catalog/descs/factory.go` line 52 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/88875#-ND9vfUnAcXXgfK-EZWY:-NDC4NPL68T2auvjKUO5:bss4a2b) ([raw file](https://github.com/cockroachdb/cockroach/blob/b46b350b61c8c938820c25ccd62c2fc8ea0694e1/pkg/sql/catalog/descs/factory.go#L52)):*
<details><summary><i>Previously, ajwerner wrote…</i></summary><blockquote>

Why do these exist? The places they seem to be accessed already have the 
</blockquote></details>

Removed.
___
*[`pkg/sql/catalog/descs/factory.go` line 58 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/88875#-ND9w95H5pfDwzU6Ed-0:-NDC4Owc69ByS9WeTOsG:bgrrl4b) ([raw file](https://github.com/cockroachdb/cockroach/blob/b46b350b61c8c938820c25ccd62c2fc8ea0694e1/pkg/sql/catalog/descs/factory.go#L58)):*
<details><summary><i>Previously, ajwerner wrote…</i></summary><blockquote>

Let's rename this to just `InternalExecutorFactory` and embed `sqlutils.InternalExecutorFactory` in it? Alternatively, what do you think about `TxnManager` as the name?
</blockquote></details>

Renamed to `TxnManager`.
___
*[`pkg/sql/catalog/descs/factory.go` line 59 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/88875#-ND9vZf50UkroTxdEMDq:-NDC4RicAmaduDmGKyJG:bzi1hnh) ([raw file](https://github.com/cockroachdb/cockroach/blob/b46b350b61c8c938820c25ccd62c2fc8ea0694e1/pkg/sql/catalog/descs/factory.go#L59)):*
<details><summary><i>Previously, ajwerner wrote…</i></summary><blockquote>

Are either of these methods needed anymore?
</blockquote></details>

Nope, removed.
___
*[`pkg/sql/catalog/descs/system_table.go` line 46 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/88875#-ND9vu3GCIEmAMQA6rnl:-NDC4Vav4kxJDc6zxsKc:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/b46b350b61c8c938820c25ccd62c2fc8ea0694e1/pkg/sql/catalog/descs/system_table.go#L46)):*
<details><summary><i>Previously, ajwerner wrote…</i></summary><blockquote>

let's change the dependency here 
</blockquote></details>

Done.
___
*[`pkg/sql/stats/automatic_stats.go` line 354 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/88875#-ND9wL3W9LYQd19yN6U5:-NDC4acn5GQBkz1ZJS5L:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/b46b350b61c8c938820c25ccd62c2fc8ea0694e1/pkg/sql/stats/automatic_stats.go#L354)):*
<details><summary><i>Previously, ajwerner wrote…</i></summary><blockquote>

let's change the dependency here to `descs.InternalExecutorFactory`
</blockquote></details>

Done.

**Review by ajwerner — APPROVED** [review]
:lgtm: just one more dep to clean up 

> Reviewed 1 of 5 files at r1, 1 of 32 files at r5, 75 of 75 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/88875#-:-NDT_tDFBQKKBdaAW_Wr:b1yaocm)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @ajwerner, @herkolategan, @rhu713, and @ZhouXing19)

___
*[`pkg/sql/catalog/descs/factory.go` line 52 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/88875#-NDTX7ncDiVxHpaygSXS:-NDTX7ncDiVxHpaygSXT:bs9uxqc) ([raw file](https://github.com/cockroachdb/cockroach/blob/d2ce5bdcbdb915ddac146677cf8fd3a8d1a80ac9/pkg/sql/catalog/descs/factory.go#L52)):*
> ```Go
> 	sqlutil.InternalExecutorFactory
> 
> 	DescsTxnWithExecutor(
> ```

nit: put some commentary on these methods -- at least referencing the sqlutil methods they are like 
___
*[`pkg/sql/stats/stats_cache.go` line 78 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/88875#-NDTXRryA7RXxsF_zSuC:-NDTXRryA7RXxsF_zSuD:b9cwczm) ([raw file](https://github.com/cockroachdb/cockroach/blob/d2ce5bdcbdb915ddac146677cf8fd3a8d1a80ac9/pkg/sql/stats/stats_cache.go#L78)):*
> ```Go
> 
> 	// Used to resolve descriptors.
> 	collectionFactory *descs.CollectionFactory
> ```

I think you can remove this dependency now

**Review by ZhouXing19 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/88875)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @ajwerner, @herkolategan, @rhu713, and @ZhouXing19)

___
*[`pkg/sql/catalog/descs/factory.go` line 52 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/88875#-NDTX7ncDiVxHpaygSXS:-NDTvBSy1wn1uOGlxckT:bdjldd1) ([raw file](https://github.com/cockroachdb/cockroach/blob/d2ce5bdcbdb915ddac146677cf8fd3a8d1a80ac9/pkg/sql/catalog/descs/factory.go#L52)):*
<details><summary><i>Previously, ajwerner wrote…</i></summary><blockquote>

nit: put some commentary on these methods -- at least referencing the sqlutil methods they are like 
</blockquote></details>

Done adding the comments.
___
*[`pkg/sql/stats/stats_cache.go` line 78 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/88875#-NDTXRryA7RXxsF_zSuC:-NDTvE2t0W5rlL3M99CH:bs7wumx) ([raw file](https://github.com/cockroachdb/cockroach/blob/d2ce5bdcbdb915ddac146677cf8fd3a8d1a80ac9/pkg/sql/stats/stats_cache.go#L78)):*
<details><summary><i>Previously, ajwerner wrote…</i></summary><blockquote>

I think you can remove this dependency now
</blockquote></details>

Done.

**Comment by ZhouXing19** [comment]
TFTR!
bors r=ajwerner

**Pull request merged**